lib/common/variadic/uint32OrHash.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package variadic
+
+import (
+	"encoding/binary"
+	"errors"
+	"io"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+)
+
+// Uint32OrHash represents a variadic type that is either uint32 or common.Hash.
+type Uint32OrHash struct {
+	value interface{}
+}
+
+// NewUint32OrHash returns a new variadic.Uint32OrHash given an int, uint32, or Hash
+func NewUint32OrHash(value interface{}) (*Uint32OrHash, error) {
+	switch v := value.(type) {
+	case int:
+		return &Uint32OrHash{
+			value: uint32(v),
+		}, nil
```

Comment:
remove?

```suggestion
```

Reply:
I think it's kinda nice to have this to we can just call `NewUint32OrHash(1)` instead of `NewUint32OrHash(uint32(1))`, what do you think?

Reply:
Ah that's true, can you add a comment about it then? It confused me initially :smile:

Reply:
Comment added whilst I was rebasing it.